h/services/annotation_moderation.py

```diff
@@ -78,6 +102,129 @@ def update_status(self, action: AnnotationAction, annotation: Annotation) -> Non
+        new_status = moderation_log.new_moderation_status
+
+        # These are the transitions that will trigger an email to be sent
+        email_sending_status_changes = {
+            (ModerationStatus.PENDING, ModerationStatus.APPROVED),
+            (ModerationStatus.PENDING, ModerationStatus.DENIED),
+            (ModerationStatus.APPROVED, ModerationStatus.PENDING),
+            (ModerationStatus.APPROVED, ModerationStatus.DENIED),
+            (ModerationStatus.DENIED, ModerationStatus.APPROVED),
+            (ModerationStatus.SPAM, ModerationStatus.APPROVED),
+        }
+        if (old_status, new_status) not in email_sending_status_changes:
+            return
+
+        template_base = "h:templates/emails/moderated_annotation_notification"
+        context = {
+            "user_display_name": author.display_name or f"@{author.username}",
+            "annotation_url": links.incontext_link(request, annotation)
+            or request.route_url("annotation", id=annotation.id),
+            "annotation": annotation,
+            "annotation_quote": annotation.quote,
+            "app_url": request.registry.settings.get("h.app_url"),
+            "unsubscribe_url": request.route_url(
+                "unsubscribe",
+                token=self._subscription_service.get_unsubscribe_token(
+                    user_id=author.userid, type_=Subscriptions.Type.MODERATED
+                ),
+            ),
+            "preferences_url": request.route_url("account_notifications"),
+            "status_change_description": self.email_status_change_description(
+                group.name, new_status
+            ),
+        }
+        email_data = EmailData(
+            recipients=[author.email],
+            subject=self.email_subject(group.name, new_status),
+            body=render(f"{template_base}.txt.jinja2", context, request=request),
+            html=render(f"{template_base}.html.jinja2", context, request=request),
+            tag=EmailTag.MODERATED,
+            subaccount=self._email_subaccount,
+        )
+        task_data = TaskData(
+            tag=email_data.tag,
+            sender_id=author.id,
+            recipient_ids=[author.id],
+            extra={"annotation_id": annotation.id},
+        )
+        try:
+            email.send.delay(asdict(email_data), asdict(task_data))
+        except OperationalError as err:  # pragma: no cover
+            report_exception(err)
+
+    @staticmethod
+    def email_subject(group_name: str, new_status: ModerationStatus) -> str:
+        """Generate the email subject based on the moderation status change."""
+        if new_status == ModerationStatus.DENIED:
+            return f"Your comment in {group_name} has been declined"
+
+        if new_status == ModerationStatus.APPROVED:
+            return f"Your comment in {group_name} has been approved"
+
+        if new_status == ModerationStatus.PENDING:
+            return f"Your comment in {group_name} is pending approval"
+
```

Comment:
Should we say "comment" or "annotation"?

I know we use the "comment" terminology in other places, but maybe it's worth discussing with the wider group. It should not block merging this PR though.